coreplugins/cesiumion/public/components/UploadDialog.jsx

```diff
@@ -0,0 +1,222 @@
+
+		this.mergedInitialValues = {
+			...UploadDialog.defaultProps.initialValues,
+			...this.props.initialValues
+		};
+
+		this.state = {
+			title : props.title,
+			...this.mergedInitialValues
+		}
+	}
+
+    show(){
+        this.dialog.show();
+    }
+
+    handleChange = (e) => {
+        const { value, name } = e.target;
+		
+		if (name === "options.textureFormat")
+		{
+			let options = {...this.state.options};
+			options["textureFormat"] = value === "Yes";
+			this.setState({ options });
+		}
+		else if (name === "options.baseTerrainId")
+		{
+			let options = {...this.state.options};
+			options["baseTerrainId"] = value;
+			this.setState({ options });
+		}
+		else
+		{
+			this.setState({ [name]: value });
+		}
+    }
+
+	handleError = msg => error => {
+		this.props.onError(msg);
+		console.error(error);
+	};
+
+	onSubmit = values => {
+		const { asset, onSubmit } = this.props;
+		values = JSON.parse(JSON.stringify(this.state));
+		const { options = {} } = values;
+
+		switch (UploadDialog.AssetSourceType[asset]) {
+			case SourceType.RASTER_TERRAIN:
+				if (options.baseTerrainId === "")
+					delete options["baseTerrainId"];
+				else options.baseTerrainId = parseInt(options.baseTerrainId);
+				options.toMeters = 1;
+				options.heightReference = "WGS84";
+				options.waterMask = false;
+				break;
+			case SourceType.CAPTURE:
+				options.textureFormat = options.textureFormat ? "KTX2" : "AUTO";
+				break;
+		}
+
+		onSubmit(values);
+	};
+
+	getSourceFields() {
+		switch (UploadDialog.AssetSourceType[this.props.asset]) {
+			case SourceType.RASTER_TERRAIN:
+				let loadOptions = ({ isLoading, isError, data }) => {
+					if (isLoading || isError)
+						return <option disabled>LOADING...</option>;
+					let userItems = data.items
+						.filter(item => item.type === "TERRAIN")
+						.map(item => (
+							<option key={item.id} value={item.id}>
+								{item.name}
+							</option>
+						));
+					return [
+						<option key={"mean-sea-level"} value={""}>
+							Mean Sea Level
+						</option>,
+						...userItems
+					];
+				};
+
+				return (
+					<BootstrapField
+						name={"options.baseTerrainId"}
+						label={"Base Terrain: "}
+						type={"select"}
+						value={this.state.options.baseTerrainId}
+						onChange={this.handleChange}
+					>
+						<IonFetcher
+							path="assets"
+							onError={this.handleError(
+								"Failed to load terrain options. " +
+									"Please check your token!"
+							)}
+						>
+							{loadOptions}
+						</IonFetcher>
+					</BootstrapField>
+				);
+			case SourceType.CAPTURE:
+				return (
+					<BootstrapField
+						name={"options.textureFormat"}
+						label={"Use KTX2 Compression"}
+						type={"select"}
+						value={this.state.options.textureFormat ? "Yes" : "No"}
+						help={
+							"KTX v2.0 is an image container format that supports Basis Universal supercompression. " +
+							"Use KTX2 compression to create a smaller tileset with better streaming performance."
```

Comment:
Same thing here can remove concatenation

Reply:
This is for code formatting/styling, makes it easier to view on the IDE/GitHub without needing to scroll horizontally.